### ISSUE TITLE
Bug in right nav styles on safari v17.1

### ISSUE THREAD

**Issue by tlambert03** [squidfunk/mkdocs-material#6281]
### Bug description

commit 521f7272ee266526102261933e383e6f74eed062 seems to have introduced a bug wherein the right nav appears clipped in safari (Version 17.1 (19616.2.9.11.7)) after page refresh.  resizing the window across any breakpoint fixes it)

https://github.com/squidfunk/mkdocs-material/assets/1609449/916443f0-c595-412f-9a87-350ab972781c

The issue is visible as of v9.4.2 and git bisect put the issue at 521f7272ee266526102261933e383e6f74eed062

chrome, Firefox, edge work fine.

### Related links

- This is the same as #6273 which was closed

I was unable to find any other related issues since sept 23, which is when 521f7272ee266526102261933e383e6f74eed062 was merged

### Reproduction

[9.4.7-right-nav.zip](https://github.com/squidfunk/mkdocs-material/files/13243633/9.4.7-right-nav.zip)

(edited to a more minimal lockfile)

### Steps to reproduce

This bug can be seen on a brand new install. with only the theme added mkdocs.yml
```sh
mkdir test
cd test
pip install --upgrade --force-reinstall mkdocs-material
mkdocs new .
echo "\ntheme:\n  name: material" >> mkdocs.yml
mkdocs serve
```

In safari v17.1 the right nav is missing until you resize the window (see example above). 

```
pip install 'mkdocs-material<9.4.2'
mkdocs serve
```

fixed...

```
pip install 'mkdocs-material==9.4.2'
mkdocs serve
```

broken

### Browser

Safari

### Before submitting

- [X] I have read and followed the [bug reporting guidelines](https://squidfunk.github.io/mkdocs-material/contributing/reporting-a-bug/).
- [X] I have attached links to [the documentation](https://squidfunk.github.io/mkdocs-material/), and possibly related [issues](https://github.com/squidfunk/mkdocs-material/issues) and [discussions](https://github.com/squidfunk/mkdocs-material/discussions).
- [X] I assure that I have [removed all customizations](https://squidfunk.github.io/mkdocs-material/contributing/reporting-a-bug/#remove-customizations) before submitting this bug report.
- [X] I have attached a __.zip file__ with a [minimal reproduction](https://squidfunk.github.io/mkdocs-material/guides/creating-a-reproduction/).

**Comment by squidfunk** [comment]
Thanks for reporting. This issue is related to #3396 and has coincidentally been fixed in #6351 solving #6344 by the awesome @alexvoss. If I revert the fix, I can observe what is described in OP, but the latest master doesn't exhibit this behavior. Thus, we can conclude that this issue is resolved.

**Comment by squidfunk** [comment]
Released as part of [9.4.10](https://squidfunk.github.io/mkdocs-material/changelog/#9.4.10).

**Comment by tlambert03** [comment]
Thanks!
